pkg/collect/host_run.go

```diff
@@ -66,14 +85,94 @@ func (c *CollectHostRun) Collect(progressChan chan<- interface{}) (map[string][]
 		return nil, errors.Wrap(err, "failed to marshal run host result")
 	}
 
-	output := NewResult()
 	output.SaveResult(c.BundlePath, resultInfo, bytes.NewBuffer(b))
 	output.SaveResult(c.BundlePath, result, bytes.NewBuffer(stdout.Bytes()))
+	// walkthrough the output directory and save result for each file
+	if runHostCollector.OutputDir != "" {
+		runInfo.OutputDir = runHostCollector.OutputDir
+		bundleOutputRelativePath = filepath.Join(collectorRelativePath, runHostCollector.OutputDir)
+		output.SaveResults(c.BundlePath, bundleOutputRelativePath, cmdOutputTempDir)
+	}
+
+	return output, nil
+}
+
+func (c *CollectHostRun) processEnvVars(cmd *exec.Cmd) (string, error) {
+	var (
+		cmdOutputTempDir string
+		err              error
+	)
+	// clears the parent env vars
+	cmd.Env = []string{}
+	guaranteedEnvs := []string{"PATH", "KUBECONFIG"}
+	for _, key := range guaranteedEnvs {
+		guaranteedEnvVal, found := os.LookupEnv(key)
+		if found {
+			cmd.Env = append(cmd.Env,
+				fmt.Sprintf("%s=%s", key, guaranteedEnvVal))
+		}
+	}
+
+	runHostCollector := c.hostCollector
+	if runHostCollector.InheritParentEnvs {
+		if runHostCollector.InheritEnvs != nil {
+			for _, key := range runHostCollector.InheritEnvs {
+				envVal, found := os.LookupEnv(key)
+				if !found {
+					return "", errors.New(fmt.Sprintf("inherit env variable is not found: %s", key))
+				}
+				cmd.Env = append(cmd.Env,
+					fmt.Sprintf("%s=%s", key, envVal))
+			}
+		} else {
+			cmd.Env = append(cmd.Env, os.Environ()...)
+		}
+	}
+
+	if runHostCollector.Env != nil {
+		for i := range runHostCollector.Env {
+			parts := strings.Split(runHostCollector.Env[i], "=")
+			if len(parts) == 2 && parts[0] != "" && parts[1] != "" {
+				cmd.Env = append(cmd.Env,
+					fmt.Sprintf("%s=%s", runHostCollector.Env[i], os.Getenv(runHostCollector.Env[i])))
+			} else {
+				return "", errors.New(fmt.Sprintf("env variable entry is missing '=' : %s", runHostCollector.Env[i]))
+			}
+		}
+	}
 
-	runHostOutput := map[string][]byte{
-		resultInfo: b,
-		result:     stdout.Bytes(),
+	// if we choose to save result for the command run
+	if runHostCollector.OutputDir != "" {
+		cmdOutputTempDir, err = os.MkdirTemp("", runHostCollector.OutputDir)
+		defer os.RemoveAll(cmdOutputTempDir)
+		if err != nil {
+			return "", errors.New(fmt.Sprintf("failed to created temp dir for: %s", runHostCollector.OutputDir))
+		}
+		cmd.Env = append(cmd.Env,
+			fmt.Sprintf("TS_WORKSPACE_DIR=%s", cmdOutputTempDir),
+		)
+	}
+
+	if runHostCollector.Input != nil {
+		cmdInputTempDir, err := os.MkdirTemp("", "input")
+		defer os.RemoveAll(cmdInputTempDir)
```

Comment:
This will get deleted as well

Reply:
moved out from the func. realized this cannot be easily moved into the function due to `defer`